src/db/databaseSource.ts

```diff
@@ -276,21 +287,37 @@ export class DatabaseSource {
         ...(dbData.ssl ? { ssl: dbData.ssl } : ''),
       },
       useNullAsDefault: true,
-      pool: {
-        min: 5,
-        max: 30,
-        createTimeoutMillis: 3000,
-        acquireTimeoutMillis: 30000,
-        idleTimeoutMillis: 30000,
-        reapIntervalMillis: 1000,
-        createRetryIntervalMillis: 100,
-        propagateCreateError: false,
-      },
+      pool: this.connectionPool(dbData),
       ...dbData.options,
     };
     return knex(options);
   }
 
+  private connectionPool(dbData: DbData) {
+    if (dbData.healthCheckConnection) {
+      return {
+        min: 1,
+        max: 2,
+        createTimeoutMillis: 500,
+        acquireTimeoutMillis: 500,
+        idleTimeoutMillis: 500,
+        reapIntervalMillis: 500,
+        createRetryIntervalMillis: 100,
+        propagateCreateError: false,
+      }
+    }
+    return {
+      min: 5,
+      max: 30,
+      createTimeoutMillis: 3000,
+      acquireTimeoutMillis: 30000,
```

Comment:
I'm not sure if those params are ok...maybe we should decrease the max connections and timeouts?